src/BoundingBoxCameraSensor.cc

```diff
@@ -0,0 +1,525 @@
+
+  this->dataPtr->imagePublisher =
+    this->dataPtr->node.Advertise<ignition::msgs::Image>(
+      this->dataPtr->topicImage);
+
+  if (this->dataPtr->type == rendering::BoundingBoxType::Box3D)
+  {
+    this->dataPtr->boxesPublisher =
+      this->dataPtr->node.Advertise<
+      ignition::msgs::AnnotatedOriented3DBox_V>(
+      this->dataPtr->topicBoundingBoxes);
+  }
+  else
+  {
+    this->dataPtr->boxesPublisher =
+      this->dataPtr->node.Advertise<
+      ignition::msgs::AnnotatedAxisAligned2DBox_V>(
+      this->dataPtr->topicBoundingBoxes);
+  }
+
+  if (!this->dataPtr->imagePublisher)
+  {
+    ignerr << "Unable to create publisher on topic ["
+      << this->dataPtr->topicImage << "].\n";
+    return false;
+  }
+
+  if (!this->dataPtr->boxesPublisher)
+  {
+    ignerr << "Unable to create publisher on topic ["
+      << this->dataPtr->topicBoundingBoxes << "].\n";
+    return false;
+  }
+
+  if (!this->AdvertiseInfo())
+    return false;
+
+  if (this->Scene())
+  {
+    this->CreateCamera();
+  }
+
+  this->dataPtr->sceneChangeConnection =
+    RenderingEvents::ConnectSceneChangeCallback(
+    std::bind(&BoundingBoxCameraSensor::SetScene, this,
+    std::placeholders::_1));
+
+  this->dataPtr->initialized = true;
+
+  return true;
+}
+
+/////////////////////////////////////////////////
+void BoundingBoxCameraSensor::SetScene(
+  ignition::rendering::ScenePtr _scene)
+{
+  std::lock_guard<std::mutex> lock(this->dataPtr->mutex);
+
+  // APIs make it possible for the scene pointer to change
+  if (this->Scene() != _scene)
+  {
+    this->dataPtr->boundingboxCamera = nullptr;
+    this->dataPtr->rgbCamera = nullptr;
+    RenderingSensor::SetScene(_scene);
+
+    if (this->dataPtr->initialized)
+      this->CreateCamera();
+  }
+}
+
+/////////////////////////////////////////////////
+bool BoundingBoxCameraSensor::CreateCamera()
+{
+  auto sdfCamera = this->dataPtr->sdfSensor.CameraSensor();
+  if (!sdfCamera)
+  {
+    ignerr << "Unable to access camera SDF element\n";
+    return false;
+  }
+
+  // Camera Info Msg
+  this->PopulateInfo(sdfCamera);
+
+  if (!this->dataPtr->rgbCamera)
+  {
+    // Create rendering camera
+    this->dataPtr->boundingboxCamera =
+      this->Scene()->CreateBoundingBoxCamera(this->Name());
+
+    this->dataPtr->rgbCamera = this->Scene()->CreateCamera(
+      this->Name() + "_rgbCamera");
+  }
+
+  auto width = sdfCamera->ImageWidth();
+  auto height = sdfCamera->ImageHeight();
+
+  // Set Camera Properties
+  this->dataPtr->rgbCamera->SetImageFormat(rendering::PF_R8G8B8);
+  this->dataPtr->rgbCamera->SetImageWidth(width);
+  this->dataPtr->rgbCamera->SetImageHeight(height);
+  this->dataPtr->rgbCamera->SetVisibilityMask(sdfCamera->VisibilityMask());
+  this->dataPtr->rgbCamera->SetNearClipPlane(sdfCamera->NearClip());
+  this->dataPtr->rgbCamera->SetFarClipPlane(sdfCamera->FarClip());
+  this->dataPtr->rgbCamera->SetNearClipPlane(0.01);
+  this->dataPtr->rgbCamera->SetFarClipPlane(1000);
```

Comment:
```suggestion
  this->dataPtr->rgbCamera->SetNearClipPlane(sdfCamera->NearClip());
  this->dataPtr->rgbCamera->SetFarClipPlane(sdfCamera->FarClip());
```
it looks like you're overriding the sdf parameters with hardcoded values. We should stick with the sdf parameters